wolfcrypt/test/test.c

```diff
@@ -13556,9 +13560,50 @@ static wc_test_ret_t simple_mem_test(int sz)
 }
 #endif
 
+/* If successful, returns the first letter of the byte array `in`.
+**
+** This is a deceptively simple test of a read-only embedded Linux file system.
+** (e.g CFLAGS `-mfdpic` and `-mforce-l32`) When `-mforce-l32` is missing,
+** access to `in` will fail as the const is on the R/O memory-mapped file
+** system. Const is *not* loaded into app memory. See PR 6523.
+**
+** Edit with caution. */
+static wc_test_ret_t const_byte_ptr_test(const byte* in, word32 *outJ)
```

Comment:
This code is still too complex and too verbose (logging wise). You can run the entire test with a single code line right?

Reply:
I did try pruning it down to something smaller. The compiler seems to optimize it and the `Illegal Instruction` is not caught during this test. I'll see what I can do here.

Reply:
Does this not work? `if (const_byte_array[0] != CBPTR_EXPECTED)`

Reply:
correct: I've confirmed that does not properly detect the problem. I'll add a comment as such.